src/inmanta/server/services/projectservice.py

```diff
@@ -0,0 +1,111 @@
+"""
+    Copyright 2019 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import asyncio
+import logging
+import uuid
+from typing import List, Optional, cast
+
+import asyncpg
+
+from inmanta import data
+from inmanta.protocol import methods
+from inmanta.protocol.exceptions import NotFound, ServerError
+from inmanta.server import SLICE_AGENT_MANAGER, SLICE_DATABASE, SLICE_PROJECT, SLICE_RESOURCE, protocol
+from inmanta.server.agentmanager import AgentManager
+from inmanta.server.services.resourceservice import ResourceService
+from inmanta.types import Apireturn
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ProjectService(protocol.ServerSlice):
+    """Slice with project and environment management"""
+
+    agentmanager: AgentManager
+    resource_service: ResourceService
+
+    def __init__(self) -> None:
+        super(ProjectService, self).__init__(SLICE_PROJECT)
+
+    def get_dependencies(self) -> List[str]:
+        return [SLICE_DATABASE, SLICE_RESOURCE]
+
+    async def prestart(self, server: protocol.Server) -> None:
+        self.agentmanager = cast(AgentManager, server.get_slice(SLICE_AGENT_MANAGER))
```

Comment:
SLICE_AGENT_MANAGER is not in dependency list.

Reply:
done